### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by mcmcgrath13**
* add hosting, analytics
* preview on PR, deploy on merge

### CODE REVIEW

File: .github/workflows/firebase-hosting-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  build_and_preview:
+    if: '${{ github.event.pull_request.head.repo.full_name == github.repository }}'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Reconfigure git to use HTTP authentication
+        run: >
+          git config --global url."https://github.com/".insteadOf
+          ssh://git@github.com/
```
Comment (eldu) [inline .github/workflows/firebase-hosting-pull-request.yml:15]: What was the reasoning behind doing this step? Just curious and trying to see if should do this for other github workflows!

**Review by eldu — APPROVED** [review]
The preview is works well! I also spun up the website locally to test out too. :) These firebase previews are really clutch for prs and such. Thanks so much! 

🎉

**Review by RobertGemmaJr — APPROVED** [review]
Don't know much about firebase but everything spins up on my end as well. Looks great! 🚀

**Review by mirestrepo — COMMENTED** [review]
🚀

**Review by kmilo9999 — APPROVED** [review]

**Comment by mirestrepo** [comment]
I'm taking the liberty of merging, so other can start benefitting from the preview-urls... I think we can still have inline conversations after, @mcmcgrath13 I'm also curios about the https/ssh part of the action I had not seen that done... was that done by Firebase CLI?

**Comment by mcmcgrath13** [comment]
The github runner doesn't play nicely with ssh github package links (see [failed action](
https://github.com/ridatadiscoverycenter/react-aframe-volume-renderer/runs/3632925375?check_suite_focus=true)), so this forces it to use https.

**Pull request merged**